File: Remove-Duplicates-from-Sorted-List.py
1# Definition for singly-linked list.
2# class ListNode(object):
3#     def __init__(self, val=0, next=None):
4#         self.val = val
5#         self.next = next
6class Solution(object):
7    def deleteDuplicates(self, head):
8        """
9        :type head: Optional[ListNode]
10        :rtype: Optional[ListNode]
11        """
12        current = head
13        
14        while current and current.next:
15            if current.val == current.next.val:
16                current.next = current.next.next
17            else:
18                current = current.next
19                
20        return head
21        

File: Remove-Duplicates-from-Sorted-List.cpp
1/**
2 * Definition for singly-linked list.
3 * struct ListNode {
4 *     int val;
5 *     ListNode *next;
6 *     ListNode() : val(0), next(nullptr) {}
7 *     ListNode(int x) : val(x), next(nullptr) {}
8 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
9 * };
10 */
11class Solution {
12public:
13    ListNode* deleteDuplicates(ListNode* head) {
14        ListNode* temp = head;
15
16        while (temp != nullptr && temp->next != nullptr) {
17            if (temp->val == temp->next->val) {
18                temp->next = temp->next->next;
19            } else {
20                temp = temp->next;
21            }
22        }
23
24        return head;
25    }
26};